internal/migrations/20210713123129_populate_infra_env.go

```diff
@@ -31,7 +31,7 @@ func populateInfraEnv() *gormigrate.Migration {
 				return err
 			}
 
-			dbClusters, err := common.GetClustersFromDBWhere(tx, common.UseEagerLoading, common.IncludeDeletedRecords)
+			dbClusters, err := common.GetClustersFromDBWhere(tx, common.SkipEagerLoading, common.IncludeDeletedRecords)
```

Comment:
How much work/time would it take to move this away from using `GetClustersFromDBWhere` entirely?

It's tough to see someone getting back to fixing this "properly" without running into another issue unless you want to do it in a follow up now?

Reply:
I can't answer that. I have no connection to the origin of this migration and apparently I was not able to find out anyone who has more context than us here after reading through the code.

Reply:
I did a quick grep and it looks like we are not using it in many places:

```
internal/bminventory/inventory.go
2999:    dbClusters, err := common.GetClustersFromDBWhere(db, common.UseEagerLoading,

internal/cluster/common_test.go
236:     clusters, err := common.GetClustersFromDBWhere(db, common.UseEagerLoading, true, "")
254:     clusters, err := common.GetClustersFromDBWhere(db, common.UseEagerLoading, true, "")
272:     clusters, err := common.GetClustersFromDBWhere(db, common.UseEagerLoading, true, "")
290:     clusters, err := common.GetClustersFromDBWhere(db, common.SkipEagerLoading, true, "")

internal/common/db.go
203:func GetClustersFromDBWhere(db *gorm.DB, eagerLoading EagerLoadingState, includeDeleted DeleteRecordsState, where ...interface{}) ([]*Cluster, error) {

internal/migrations/20210713123129_populate_infra_env.go
34:          dbClusters, err := common.GetClustersFromDBWhere(tx, common.UseEagerLoading, common.IncludeDeletedRecords)

internal/migrations/20210822134659_multiple_networks.go
12:      dbClusters, err := common.GetClustersFromDBWhere(tx, common.UseEagerLoading, common.IncludeDeletedRecords)
```

We should be able to write an equivalent of this function to be used during migrations. It's important to avoid using any of the internal structures since those may change in the future.

Let's do this in a follow-up PR, tho.